ntoskrnl/mm/ARM3/sysldr.c

```diff
@@ -1503,7 +1503,7 @@ MiFindInitializationCode(OUT PVOID *StartVa,
             InitFound = FALSE;
 
             /* Is this the INIT section or a discardable section? */
-            if ((*(PULONG)Section->Name == 'TINI') ||
+            if ((strncmp((PCCH)Section->Name, "INIT", 5) == 0) ||
```

Comment:
Shouldn't it be "4" instead of "5" ?

Reply:
If it was 4, it would stop after comparing 4 chars and would match "INITALY" etc. as well.

Reply:
ok I didn't realize that Section->Name is a NULL-terminated string.

Reply:
Right, but then you might as well use strcmp

Reply:
From the PE specification: "An 8-byte, null-padded UTF-8 encoded string. If the string is exactly 8 characters long, there is no terminating null. "

Reply:
What? `strcmp(x, "INIT", 5)` is just a more complicated way to write `strcmp(x, "INIT")` -- the two behave exactly the same.